src/structarray.jl

```diff
@@ -512,6 +515,12 @@ function BroadcastStyle(b::AbstractArrayStyle{M}, a::StructArrayStyle{S, N}) whe
     S′ = Broadcast.result_style(S(), b)
     return S′ isa StructArrayStyle ? typeof(S′)(Val{N′}()) : StructArrayStyle{typeof(S′), N′}()
 end
+function BroadcastStyle(
+    ::Union{SparseArrays.HigherOrderFns.SparseMatStyle, SparseArrays.HigherOrderFns.SparseVecStyle},
```

Comment:
The aqua addition looks good to me. CI is all red here, even when it was green on master Saying something about "SparseArrays" not found. That should probably get fixed. Maybe this line?

I have not reviewed the changes to src/structarray.jl.

Reply:
should I import SparseArrays here?

Reply:
Seems like a bad idea to add SparseArrays as a dependency just for this. Maybe a weak dependency if necessary?

Reply:
There is a `SparseArrays` extension now, so this may be moved to that?